docs/about/themes-translation.md

```diff
@@ -0,0 +1,195 @@
+# MkDocs themes translation
+
+How to add a locale translation to MkDocs built-in themes.
+
+---
+
+## Localization tooling prerequisites
+
+Theme localization uses the [babel][babel] project to handle theme localization
+files generation and compilation. We worked on wrapping its usage through
+`setup.py` commands as described below to make updating and contributing
+translations easy.
+
+Make sure they are installed in your environment:
+
+```bash
+pip install -r requirements/project.txt
+```
+
+[babel]: http://babel.pocoo.org/en/latest/cmdline.html
+
+## Contributing theme translations
+
+If your favorite language locale is not supported yet on the `mkdocs` and
+`readthedocs` built-in themes, you can easily contribute it to the project by
+following the steps below.
+
+### Creating the theme localization files
+
+Theme localization is achieved by *extracting text* to be translated from HTML
+theme files, then *initializing* a `messages.po` file for a given locale and
+finally *adding translations* into it.
+
+Available translation locales are listed under the `locales` folder of every
+supported theme:
+
+```text
+mkdocs/themes/mkdocs/locales
+└── fr
+    └── LC_MESSAGES
+        └── messages.po
+```
+
+Say you'd like to add theme localization for the `es` (Spanish) language, you
+would follow the steps below:
+
+#### Extracting text from themes
+
+This step will parse and extract the text from the HTML sources of the themes:
```

Comment:
This section needs to be moved. Extracting is not something a translator needs to do as the POT file already exists. Instead, this step needs to happen any time a change is made to the templates. We have two options:

1. We could require the contributor (whoever submits a PR) to regenerate the POT file by running the extract command.
2. We could automate it in the GitHub workflow for a PR by checking for edits to the theme dirs and if any were made, running the extract command and updating the PR to include the new POT file. Ideally, the action would be smart enough to only run on the theme that was edited.

Reply:
As someone with a lot of experience with GitHub workflows... pushing like that is almost never done. It's way too confusing.

But *do* add the generation step to the workflow, just assert that generation doesn't cause any diff. (quite literally running `git diff --exit-code` at the end)

Reply:
Sigh. That's unfortunate. This is exactly the sort of thing a git hook is great at. But we can't expect every potential contributor to set up a local git hook. I figured a workflow would be the closest thing to a server run git hook. But, as I think about it, there is some weird gymnastics getting a workflow to have write permissions to a PR which comes from a fork. Not even core devs create PRs from this repo. We all use our own private forks.